machine.go

```diff
@@ -147,10 +147,17 @@ func (m Machine) toNode(includeRoutes bool) (*tailcfg.Node, error) {
 		keyExpiry = time.Time{}
 	}
 
+	var hostname string
+	if h.cfg.DNSConfig != nil && h.cfg.DNSConfig.Proxied { // MagicDNS
+		hostname = fmt.Sprintf("%s.%s.%s", m.Name, m.Namespace.Name, h.cfg.BaseDomain)
+	} else {
+		hostname = m.Name
+	}
+
```

Comment:
As mentioned above, this could be computed outside of this function, possible it can also be its own function.

Reply:
Does it make sense to make a function out of 5 lines? I rather leave it for the time being there.